subiquity/server/controllers/install.py

```diff
@@ -306,6 +306,36 @@ async def run_curtin_step(name, stages, step_config, source=None):
                 source=source,
                 )
             await self.setup_target(context=context)
+
+            # For OEM, we basically mimic what ubuntu-drivers does:
+            # 1. Install each package with apt-get install
+            # 2. For each package, run apt-get update using only the source
+            # installed by said package.
+            # 3. Run apt-get install again for each package. This will upgrade
+            # them to the version found in the OEM archive.
+
+            # NOTE In ubuntu-drivers, this is done in a single call to apt-get
+            # install.
+            for pkg in self.model.oem.metapkgs:
+                await self.install_package(package=pkg)
+
+            if self.model.network.has_network:
+                for pkg in self.model.oem.metapkgs:
+                    source_list = f"/etc/apt/sources.list.d/{pkg}.list"
+                    await run_curtin_command(
+                        self.app, context,
+                        "in-target", "-t", self.tpath(), "--",
+                        "apt-get", "update",
+                        "-o", f"Dir::Etc::SourceList={source_list}",
+                        "-o", "Dir::Etc::SourceParts=/dev/null",
+                        "--no-list-cleanup",
+                        private_mounts=False)
+
+                # NOTE In ubuntu-drivers, this is done in a single call to
+                # apt-get install.
```

Comment:
I think (but am not sure) that in any realistic scenario there is only one metapackage for any system. Do you know differently?

Reply:
I'm not sure. I haven't seen the scenario where we would have more than one in practice but I only played around with a single certified laptop. I guess I should add a note somewhere that only one (or zero) is expected - but I don't think we should bail if ubuntu-drivers gives us two.